pkg/storage/asset_store.go

```diff
@@ -11,21 +11,27 @@ import (
 
 // AssetStore is a wrapper around a BlobAccess to inteface well with
 // AssetReference messages
-type AssetStore struct {
+type AssetStore interface {
+	Get(ctx context.Context, ref *asset.AssetReference, instance digest.InstanceName) (*asset.Asset, error)
+	Put(ctx context.Context, ref *asset.AssetReference, data *asset.Asset, instance digest.InstanceName) error
+}
+
+// blobAccessAssetStore is an AssetStore backed by a blobAccess.
+type blobAccessAssetStore struct {
```

Comment:
This should be in its own file if we're making `AssetStore` an interface, imo, similarly the `NewAssetStore` should become `NewBlobAccessAssetStore`.

Reply:
If metrics asset store is superfluous I will instead revert those changes.

Reply:
If metrics asset store is superfluous I will instead revert those changes.